vertx-pg-client/src/main/java/io/vertx/pgclient/PgErrorCodes.java

```diff
@@ -0,0 +1,306 @@
+/*
+ * Copyright (c) 2020 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ */
+
+package io.vertx.pgclient;
+
+/**
+ * SQLSTATE codes used by PostgreSQL
+ * (<a href="https://www.postgresql.org/docs/current/errcodes-appendix.html">PostgreSQL Error Codes</a>,
+ * <a href="https://git.postgresql.org/gitweb/?p=postgresql.git;a=blob;f=src/backend/utils/errcodes.txt;hb=HEAD">
+ * errcodes.txt</a>)
+ */
+public class PgErrorCodes {
```

Comment:
I think this should be instead an enum

Reply:
I don't think so.
The list of error codes is never complete. New PostgreSQL versions may add new codes, and any application can use custom codes: `do $$ begin raise sqlstate '99999'; end $$;`
Sub-classing PgErrorCodes allows to add custom codes.
An enum cannot be extended to support more codes.
There is no need to get a human readable name for a code at runtime because PgException contains a message and a detail field. PgErrorCodes provides a human readable name for an expected code used in a unit test or when to handle a specific error.
"You should use enum types any time you need to represent a fixed set of constants." (quote from https://docs.oracle.com/javase/tutorial/java/javaOO/enum.html )

Reply:
ok, then I think we should not have this list or error codes in the code

Reply:
I've removed the file PgErrorCodes.java.